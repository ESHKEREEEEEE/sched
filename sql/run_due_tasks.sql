CREATE EXTENSION sched;

TRUNCATE sched.tasks, sched.whitelist;

INSERT INTO sched.whitelist (command) VALUES ('ls');

INSERT INTO sched.tasks (command, run_at, status, type)
VALUES ('ls', now(), 'pending', 'sh');

SELECT run_due_tasks();

SELECT id, command, status, error IS NULL AS no_error
FROM sched.tasks
ORDER BY id;

