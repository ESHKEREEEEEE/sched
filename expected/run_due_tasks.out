CREATE EXTENSION sched;
TRUNCATE sched.tasks, sched.whitelist;
INSERT INTO sched.whitelist (command) VALUES ('ls');
INSERT INTO sched.tasks (command, run_at, status, type)
VALUES ('ls', now(), 'pending', 'sh');
SELECT run_due_tasks();
 run_due_tasks 
---------------
             0
(1 row)

SELECT id, command, status, error IS NULL AS no_error
FROM sched.tasks
ORDER BY id;
 id | command | status | no_error 
----+---------+--------+----------
  1 | ls      | done   | t
(1 row)

